### PR TITLE
AEA-1383 update 'difference_to_fetched_agent' to load all pages of agent yaml config

### LINKED ISSUE #2130: difference_to_fetched_agent should consider also component overrides
**Is your feature request related to a problem? Please describe.**
In the integration test the tool `difference_to_fetched_agent` does not consider component overrides

**Describe the solution you'd like**
Extend accordingly and fix the integration tests

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

In the integration test the tool `difference_to_fetched_agent` does not consider component overrides.
Thus PR updates it accordingly.

## Fixes

Fixes #2130

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: aea/test_tools/test_cases.py [hunk 1]
```diff
@@ -297,35 +297,64 @@ def difference_to_fetched_agent(cls, public_id: str, agent_name: str) -> List[st
...
+            for key, value in content1_agentconfig_copy.items():
+                if content2_agentconfig[key] == value:
+                    content1_agentconfig.pop(key)
+                    content2_agentconfig.pop(key)
             allowed_diff_keys = [
                 "aea_version",
                 "author",
                 "description",
                 "version",
+                "connection_private_key_paths",
```
Comment (DavidMinarsch) [inline aea/test_tools/test_cases.py:326]: Why do they need to be added?

File: packages/fetchai/agents/erc1155_client/aea-config.yaml [hunk 2]
```diff
@@ -42,6 +42,11 @@ dependencies:
   aea-ledger-fetchai:
     version: <0.2.0,>=0.1.0
 ---
+public_id: fetchai/soef:0.18.0
```
Comment (DavidMinarsch) [inline packages/fetchai/agents/erc1155_client/aea-config.yaml:45]: Nice! Immediate gains :)

File: tests/test_packages/test_skills_integration/test_tac.py [hunk 3]
```diff
@@ -421,52 +415,42 @@ def test_tac(self):
         setting_path = "vendor.fetchai.skills.tac_control_contract.models.parameters.args.service_data"
         self.nested_set_config(setting_path, data)
 
-        default_routing = {
```
Comment (DavidMinarsch) [inline tests/test_packages/test_skills_integration/test_tac.py:424]: Why removed?

Comment (DavidMinarsch) [reply tests/test_packages/test_skills_integration/test_tac.py:424]: I think duplicate?

File: aea/test_tools/test_cases.py (hunk 1 above)
Comment (DavidMinarsch) [reply aea/test_tools/test_cases.py:326]: I see, just for convenience

File: tests/test_packages/test_skills_integration/test_tac.py (hunk 3 above)
Comment (marcofavorito) [reply tests/test_packages/test_skills_integration/test_tac.py:424]: yes

File: packages/fetchai/agents/erc1155_client/aea-config.yaml (hunk 2 above)
Comment (marcofavorito) [reply packages/fetchai/agents/erc1155_client/aea-config.yaml:45]: indeed

File: aea/test_tools/test_cases.py (hunk 1 above)
Comment (marcofavorito) [reply aea/test_tools/test_cases.py:326]: yes. Also, don't you think it makes more sense? O/w we could add keys after the comparison with fetched agent

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Review by jrriehl — APPROVED** [review]
Nice one @marcofavorito

**Pull request merged**